app/src/store/features/userActions.js

```diff
@@ -27,13 +27,8 @@ export const updateUsername = (prevState, action) => {
 };
 
 //Persona Survey actions
-
-export const updateUserPersona = (prevState, action) => {
-  prevState.userPersona.persona = action.payload;
-};
-
-export const updatePersonaReferralChannel = (prevState, action) => {
-  prevState.userPersona.referralChannel = action.payload;
+export const updateSingleChoicePersonaFields = (prevState, action) => {
+  prevState.userPersona[action.payload.key] = action.payload.value;
```

Comment:
This function name does not give any clarity about what this can be used for, unless someone has the whole understanding of the configs we have created for the persona workflow.